server/icon.go

```diff
@@ -0,0 +1,235 @@
+	file, header, err := r.FormFile("icon")
+	if err != nil {
+		handleErrorWithCode(logger, w, http.StatusBadRequest, "Failed to get uploaded file", err)
+		return
+	}
+	defer file.Close()
+
+	// Validate file type
+	contentType := header.Header.Get("Content-Type")
+	if !strings.HasPrefix(contentType, "image/png") {
+		http.Error(w, "Invalid file type. Must be PNG", http.StatusBadRequest)
+		return
+	}
+
+	// Read file data
+	data, err := io.ReadAll(file)
+	if err != nil {
+		handleErrorWithCode(logger, w, http.StatusInternalServerError, "Failed to read file data", err)
+		return
+	}
+
+	// Validate file size
+	if len(data) > 1024*1024 { // 1MB
+		http.Error(w, "File too large. Maximum size is 1MB", http.StatusBadRequest)
+		return
+	}
+
+	// Validate image dimensions
+	img, _, err := image.DecodeConfig(bytes.NewReader(data))
+	if err != nil {
+		http.Error(w, "Invalid PNG image", http.StatusBadRequest)
+		return
+	}
+
+	// Check if image is square and within acceptable size range
+	if img.Width != img.Height {
+		http.Error(w, "Image must be square (width and height must be equal)", http.StatusBadRequest)
+		return
+	}
+
+	if img.Width < 150 || img.Width > 300 || img.Height < 150 || img.Height > 300 {
+		http.Error(w, "Image must be between 150x150 and 300x300 pixels, and should be 192x192 pixels", http.StatusBadRequest)
+		return
+	}
+
+	// Store the icon in KV store
+	err = a.p.pluginStore.StoreIcon(string(iconType), data)
+	if err != nil {
+		handleErrorWithCode(logger, w, http.StatusInternalServerError, "Failed to store icon", err)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	response := map[string]interface{}{
+		"success":  true,
+		"iconPath": fmt.Sprintf("/plugins/com.mattermost.plugin-msteams-devsecops/icons/%s", iconType),
+	}
+	_ = json.NewEncoder(w).Encode(response)
+}
+
+// getIcon serves custom icons or falls back to default icons
+func (a *API) getIcon(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	iconTypeStr := vars["iconType"]
+	iconType := IconType(iconTypeStr)
+
+	if !iconType.IsValid() {
+		http.Error(w, "Invalid icon type", http.StatusBadRequest)
+		return
+	}
+
+	// Try to get custom icon from KV store
+	iconData, err := a.p.pluginStore.GetIcon(string(iconType))
+	if err != nil {
+		// If not found, serve default icon
+		var defaultData []byte
+		if iconType == IconTypeColor {
+			defaultData = assets.LogoColorData
+		} else {
+			defaultData = assets.LogoOutlineData
+		}
+
+		w.Header().Set("Content-Type", "image/png")
+		w.Header().Set("Cache-Control", "no-cache, no-store, must-revalidate")
+		w.Header().Set("Pragma", "no-cache")
+		w.Header().Set("Expires", "0")
+		w.WriteHeader(http.StatusOK)
+		_, _ = w.Write(defaultData)
+		return
+	}
+
+	// Serve custom icon
+	w.Header().Set("Content-Type", "image/png")
+	w.Header().Set("Cache-Control", "no-cache, no-store, must-revalidate")
+	w.Header().Set("Pragma", "no-cache")
+	w.Header().Set("Expires", "0")
+	w.WriteHeader(http.StatusOK)
```

Comment:
Can we simplify this?

```suggestion
                iconData = assets.LogoOutlineData
		if iconType == IconTypeColor {
			iconData = assets.LogoColorData
		}
	}

	// Serve custom icon
	w.Header().Set("Content-Type", "image/png")
	w.Header().Set("Cache-Control", "no-cache, no-store, must-revalidate")
	w.Header().Set("Pragma", "no-cache")
	w.Header().Set("Expires", "0")
	w.WriteHeader(http.StatusOK)
```